vars/tdr.groovy

```diff
@@ -0,0 +1,47 @@
+// Call when deployment (intg or staging) has finished. Use the delay to ensure that the AWS load balancer allows access to the new version you are deploying.
+def runEndToEndTests(int delaySeconds, String stage) {
+    String[] params = [
+            string(name: "STAGE", value: stage)
+    ]
+    build(job: "TDRAcceptanceTest", parameters: params, quietPeriod: delaySeconds, wait: false)
+}
+
+
+//It is important for TDR devs to know that the code they want to merge doesn't break TDR. By sending the build status for every commit (all branches) to GitHub we can ensure code that breaks TDR cannot be merged.
+
+// Call this when build starts (to let person who made changes know they are being checked) - call within first 'stage' of Jenkins pipeline actions.
+def reportStartOfBuildToGitHub(String repo) {
+    withCredentials([string(credentialsId: 'github-jenkins-api-key', variable: 'GITHUB_ACCESS_TOKEN')]) {
+        String sha = sh(script: 'git rev-parse HEAD', returnStdout: true).trim()
+        String url = "https://api.github.com/repos/nationalarchives/${repo}/statuses/${sha}"
+        sh "curl -XPOST '${url}' -H 'Authorization: bearer ${env.GITHUB_ACCESS_TOKEN}' --data '{\"state\":\"pending\",\"target_url\":\"${env.BUILD_URL}\",\"description\":\"Jenkins build has started\",\"context\":\"TDR Jenkins build status\"}'"
+    }
+}
+
+// Call when build finishes successfully - in Jenkins pipeline 'post' actions
+def reportSuccessfulBuildToGitHub(String repo) {
+    withCredentials([string(credentialsId: 'github-jenkins-api-key', variable: 'GITHUB_ACCESS_TOKEN')]) {
+        String sha = sh(script: 'git rev-parse HEAD', returnStdout: true).trim()
+        String url = "https://api.github.com/repos/nationalarchives/${repo}/statuses/${sha}"
+        sh "curl -XPOST '${url}' -H 'Authorization: bearer ${env.GITHUB_ACCESS_TOKEN}' --data '{\"state\":\"success\",\"target_url\":\"${env.BUILD_URL}\",\"description\":\"Jenkins build has completed successfully\",\"context\":\"TDR Jenkins build status\"}'"
+    }
+}
+
+// Call when build fails - in Jenkins pipeline 'post' actions
+def reportFailedBuildToGitHub(String repo) {
+    withCredentials([string(credentialsId: 'github-jenkins-api-key', variable: 'GITHUB_ACCESS_TOKEN')]) {
```

Comment:
The code in this three functions is very similar - it'd be worth refactoring this into a shared method so the code only has to be updated in one place.

Reply:
Sorry, I didn't explain that very well. I was trying to suggest that you kept the existing three function as well as extracting a common helper function! It goes back to the [original comment](https://github.com/nationalarchives/tdr-jenkinslib/pull/1#discussion_r427816894) about providing a clear interface to people calling this code.

Let me know if it would be easier to have chat about this. 🙂

Reply:
Ah I see! I'll revert that change. Should I extract out the section that gets the sha & creates the github url to post to the API to the helper function?

Reply:
I've changed this and added a helper function for the API URL. I also tested it out on Jenkins to make sure it didn't break the existing functions. I think it looks much better and is a bit more DRY